File: e-commerce/src/main/java/com/mehmetkerem/model/Address.java
package com.mehmetkerem.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

@Getter
@Setter
@Entity
@Table(name = "addresses")
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String country;

    private String city;

    private String district;

    @Column(name = "address_line")
    private String addressLine;

    @Column(name = "postal_code")
    private String postalCode;

    @Column(name = "user_id")
    private Long userId;
}

File: can-antika-frontend/lib/types/address.ts
export interface AddressResponse {
    id: number;
    title: string;
    country: string;
    city: string;
    district: string;
    phone?: string | null;
    postalCode: string;
    addressLine: string;
}

export interface AddressRequest {
    title: string;
    country: string;
    city: string;
    district: string;
    phone: string;
    postalCode: string;
    addressLine: string;
}
